app/src/features/projects/components/ProjectFundingItemForm.tsx

```diff
@@ -77,172 +73,118 @@ export interface IProjectFundingItemFormProps {
-                <Box mt={4}>
-                  <Divider />
-                </Box>
-              </DialogContent>
-              <DialogActions>
-                <Button
-                  variant="contained"
-                  color="primary"
-                  onClick={async () => {
-                    handleSubmit();
-                  }}
-                  autoFocus
-                  aria-label="Add Funding Source">
-                  Add
-                </Button>
-                <Button
-                  variant="outlined"
-                  onClick={() => {
-                    props.onCancel();
-                    resetForm();
-                  }}
-                  color="primary">
-                  Cancel
-                </Button>
-              </DialogActions>
-            </Dialog>
-          </>
-        );
-      }}
-    </Formik>
+                  // If an agency_id with a `Not Applicable` investment_action_category is chosen, auto select
+                  // it for the user.
+                  if (event.target.value !== 1 && event.target.value !== 2) {
+                    setFieldValue(
+                      'investment_action_category',
+                      props.investment_action_category.find((item) => item.fs_id === event.target.value)?.value || 0
+                    );
+                  }
+                }}
+                error={touched.agency_id && Boolean(errors.agency_id)}
+                displayEmpty
+                inputProps={{ 'aria-label': 'Agency Name' }}>
+                {props.funding_sources.map((item) => (
+                  <MenuItem key={item.value} value={item.value}>
+                    {item.label}
+                  </MenuItem>
+                ))}
+              </Select>
+              <FormHelperText>{errors.agency_id}</FormHelperText>
+            </FormControl>
+          </Grid>
+          {investment_action_category_label && (
+            <Grid item xs={12}>
+              <FormControl fullWidth variant="outlined" required={true} style={{ width: '100%' }}>
+                <InputLabel id="investment_action_category-label">{investment_action_category_label}</InputLabel>
+                <Select
+                  id="investment_action_category"
+                  name="investment_action_category"
+                  labelId="investment_action_category-label"
+                  label={investment_action_category_label}
+                  value={values.investment_action_category}
+                  onChange={handleChange}
+                  error={touched.investment_action_category && Boolean(errors.investment_action_category)}
+                  displayEmpty
+                  inputProps={{ 'aria-label': `${investment_action_category_label}` }}>
+                  {props.investment_action_category
+                    // Only show the investment action categories whose fs_id matches the agency_id id
+                    .filter((item) => item.fs_id === values.agency_id)
```

Comment:
can we add a test case for this filter and map case?
means we need a test case where the fs_id is equal to the agency_id so that the filter is true and then it enters the map